### DISCUSSION TITLE
AttributeError: no attribute 'add_to_pipe'

### DISCUSSION THREAD

**Discussion by pranavnbapat** [explosion/spaCy#11718] [Help: Coding & Implementations] [answered]
I followed exactly the instructions in this URL for usage:
https://spacy.io/universe/project/contextualSpellCheck

I get:
AttributeError: partially initialized module 'contextualSpellCheck' has no attribute 'add_to_pipe' (most likely due to a circular import)

Does anybody know how to solve this?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @pranavnbapat , can you post the whole traceback in your question? It seems that I can't replicate your problem. In the meantime, can you try doing this instead:

```python
import spacy
import contextualSpellCheck

nlp = spacy.load("en_core_web_sm")

# You can pass the optional parameters to the contextualSpellCheck
# eg. pass max edit distance use config={"max_edit_dist": 3}
nlp.add_pipe("contextual spellchecker")
```

Lastly, [contextualSpellCheck](https://github.com/R1j1t/contextualSpellCheck) is a third-party package from the spaCy Universe. You might be better off asking directly to the original author's [discussion forum](https://github.com/R1j1t/contextualSpellCheck/discussions).

**Reply by pranavnbapat** [reply]
Thank you for your reply.

![image](https://user-images.githubusercontent.com/7271334/199027652-e76d8e90-ca2a-4fe3-9ea7-ca7e8f5d9cc8.png)

This is what I'm doing. I also tried what you suggested, but didn't work.

And thank you for providing me with the discussion forum link. I'll also ask there. :)

**Reply by adrianeboyd** [reply]
The problem is that you've given your script the same name as the module, and python ends up trying to import your script as the module `contextualSpellCheck` instead of the installed module.

Rename your script, choosing a unique name that doesn't overlap with any module names.

**Reply by pranavnbapat** [reply]
Ohh, how stupid of me... Thank you for helping out... :)

**Comment by david-engelmann** [comment]
The spacy docs alert to a update in the declaration for 3.0 and higher. [Here is the section in the docs](https://spacy.io/usage/processing-pipelines#custom-components). I will also include a screenshot of the relevant information below.

![image](https://user-images.githubusercontent.com/81196676/200009307-845ce342-2ec1-43d1-ba39-b3d9ac539275.png)

**Reply by pranavnbapat** [reply]
Thank you david... :)
